### PR TITLE
docs: add more architecture description

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

Add corresponding explanations to the architecture diagrams in the APISIX documentation to help readers better understand them.

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/zh/latest/architecture-design/apisix.md [hunk 1]
```diff
@@ -25,6 +25,14 @@ title: APISIX
 
 ![软件架构](../../../assets/images/flow-software-architecture.png)
 
+Apache APISIX 是一个动态、实时、高性能的云原生 API 网关。它构建于 Nginx + ngx_lua 的技术基础之上，充分利用了 luajit 所提供的强大性能。
```
Comment (hf400159) [inline docs/zh/latest/architecture-design/apisix.md:28]:
```suggestion
Apache APISIX 是一个动态、实时、高性能的云原生 API 网关。它构建于 NGINX + ngx_lua 的技术基础之上，充分利用了 LuaJIT 所提供的强大性能。
```

File: docs/zh/latest/architecture-design/apisix.md [hunk 2]
```diff
@@ -25,6 +25,14 @@ title: APISIX
...
+
+APISIX 主要分为两个部分，其一是 APISIX 核心，包括 Lua 插件、多语言插件运行时、WASM 插件运行时等；其二是功能丰富的各种内置插件，包括可观测性、安全、流量控制等。
```
Comment (hf400159) [inline docs/zh/latest/architecture-design/apisix.md:30]:
```suggestion
APISIX 主要分为两个部分，其一是 APISIX 核心功能，包括 Lua 插件、多语言插件运行时、Wasm 插件运行时等；其二是功能丰富的各种内置插件，包括可观测性、安全、流量控制等。
```

File: docs/zh/latest/architecture-design/apisix.md [hunk 3]
```diff
@@ -25,6 +25,14 @@ title: APISIX
...
+
+在 APISIX 核心中，提供了路由匹配、负载均衡、服务发现、管理 API 等重要功能，以及配置管理等的基础性模块。除此之外，APISIX 插件运行时也包含其中，提供原生 Lua 插件的运行框架、多语言插件的运行框架、实验性的 WASM 插件运行时等。APISIX 多语言插件运行时提供多种不同的其他语言的支持，比如 Golang、Java、Python、JS 等。
```
Comment (hf400159) [inline docs/zh/latest/architecture-design/apisix.md:32]:
```suggestion
APISIX 在其核心功能中提供了路由匹配、负载均衡、服务发现、管理 API 等重要特性，以及配置管理等基础性模块。除此之外，APISIX 插件运行时也包含其中，提供了原生 Lua 插件和多语言插件的运行框架，以及实验性的 Wasm 插件运行时等。APISIX 多语言插件运行时提供多种开发语言的支持，比如 Golang、Java、Python、JS 等。
```

File: docs/zh/latest/architecture-design/apisix.md [hunk 4]
```diff
@@ -25,6 +25,14 @@ title: APISIX
...
+
+APISIX 也内置了各类插件，覆盖了 API 网关的各种领域，如认证鉴权、安全、可观测性、流量管理、多协议接入等。当前 APISIX 内置的插件使用原生 Lua 实现，关于它们的介绍与使用方式，可以查看它们的插件文档。
```
Comment (hf400159) [inline docs/zh/latest/architecture-design/apisix.md:34]:
```suggestion
APISIX 目前也内置了各类插件，覆盖了 API 网关的各种领域，如认证鉴权、安全、可观测性、流量管理、多协议接入等。APISIX 内置的插件使用原生 Lua 实现，关于各个插件的介绍与使用方式，可以查看相关插件的文档。
```

File: docs/en/latest/architecture-design/apisix.md [hunk 5]
```diff
@@ -25,6 +25,14 @@ title: APISIX
 
 ![flow-software-architecture](../../../assets/images/flow-software-architecture.png)
 
+Apache APISIX is a dynamic, real-time, high-performance cloud-native API gateway. It is built on top of Nginx + ngx_lua technology and leverages the power offered by luajit.
```
Comment (hf400159) [inline docs/en/latest/architecture-design/apisix.md:28]:
```suggestion
Apache APISIX is a dynamic, real-time, high-performance cloud-native API gateway. It is built on top of Nginx + ngx_lua technology and leverages the power offered by LuaJIT.
```

File: docs/en/latest/architecture-design/apisix.md [hunk 6]
```diff
@@ -25,6 +25,14 @@ title: APISIX
...
+
+APISIX is divided into two main parts, one is the APISIX core, including Lua plugin runtime, multi-language plugin runtime, WASM plug-in runtime, etc.; the other is a variety of feature-rich built-in plugins, including observability, security, traffic control, etc.
```
Comment (hf400159) [inline docs/en/latest/architecture-design/apisix.md:30]:
```suggestion
APISIX is divided into two main parts, one is the APISIX core, including Lua plugin runtime, multi-language plugin runtime, Wasm plugin runtime, etc.; the other is a variety of feature-rich built-in plugins, including observability, security, traffic control, etc.
```

**Review by hf400159 — COMMENTED** [review]
Please refer to Chinese modification.

File: docs/zh/latest/architecture-design/apisix.md (hunk 2 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md:30]: This corresponds to the `APISIX Core` in the architecture diagram.
image

File: docs/en/latest/architecture-design/apisix.md (hunk 6 above)
Comment (bzp2010) [reply docs/en/latest/architecture-design/apisix.md]: updated

File: docs/en/latest/architecture-design/apisix.md (hunk 5 above)
Comment (bzp2010) [reply docs/en/latest/architecture-design/apisix.md]: updated

File: docs/zh/latest/architecture-design/apisix.md (hunk 4 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

File: docs/zh/latest/architecture-design/apisix.md (hunk 3 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

File: docs/zh/latest/architecture-design/apisix.md (hunk 1 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

**Review by tzssangglass — DISMISSED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Review by hf400159 — APPROVED** [review]

File: docs/zh/latest/architecture-design/apisix.md [hunk 7]
```diff
@@ -25,6 +25,14 @@ title: APISIX
 
 ![软件架构](../../../assets/images/flow-software-architecture.png)
 
+Apache APISIX 是一个动态、实时、高性能的云原生 API 网关。它构建于 NGINX + ngx_lua 的技术基础之上，充分利用了 LuaJIT 所提供的强大性能。
+
+APISIX 主要分为两个部分，其一是 APISIX 核心，包括 Lua 插件、多语言插件运行时、Wasm 插件运行时等；其二是功能丰富的各种内置插件，包括可观测性、安全、流量控制等。
```
Comment (juzhiyuan) [inline docs/zh/latest/architecture-design/apisix.md:30]:
```suggestion
APISIX 主要分为两个部分：
1. APISIX 核心：包括 Lua 插件、多语言插件运行时（Plugin Runner）、Wasm 插件运行时等；
2. 功能丰富的各种内置插件：包括可观测性、安全、流量控制等插件。
```

File: docs/zh/latest/architecture-design/apisix.md [hunk 8]
```diff
@@ -25,6 +25,14 @@ title: APISIX
...
+
+APISIX 在其核心中，提供了路由匹配、负载均衡、服务发现、管理 API 等重要功能，以及配置管理等基础性模块。除此之外，APISIX 插件运行时也包含其中，提供原生 Lua 插件的运行框架和多语言插件的运行框架，以及实验性的 Wasm 插件运行时等。APISIX 多语言插件运行时提供多种开发语言的支持，比如 Golang、Java、Python、JS 等。
```
Comment (juzhiyuan) [inline docs/zh/latest/architecture-design/apisix.md:32]:
```suggestion
APISIX 在其核心中，提供了路由匹配、负载均衡、服务发现、API 管理等重要功能，以及配置管理等基础性模块。除此之外，APISIX 插件运行时也包含其中，提供原生 Lua 插件的运行框架和多语言插件的运行框架，以及实验性的 Wasm 插件运行时等。APISIX 多语言插件运行时提供多种开发语言的支持，比如 Golang、Java、Python、JS 等。
```

File: docs/zh/latest/architecture-design/apisix.md [hunk 9]
```diff
@@ -25,6 +25,14 @@ title: APISIX
...
+
+APISIX 目前也内置了各类插件，覆盖了 API 网关的各种领域，如认证鉴权、安全、可观测性、流量管理、多协议接入等。当前 APISIX 内置的插件使用原生 Lua 实现，关于各个插件的介绍与使用方式，可以查看相关插件的文档。
```
Comment (juzhiyuan) [inline docs/zh/latest/architecture-design/apisix.md:34]:
```suggestion
APISIX 目前也内置了各类插件，覆盖了 API 网关的各种领域，如认证鉴权、安全、可观测性、流量管理、多协议接入等。当前 APISIX 内置的插件使用原生 Lua 实现，关于各个插件的介绍与使用方式，可以查看相关[插件文档](https://apisix.apache.org/docs/apisix/plugins/batch-requests)。
```

File: docs/zh/latest/architecture-design/apisix.md [hunk 10]
```diff
@@ -25,6 +25,14 @@ title: APISIX
 
 ![软件架构](../../../assets/images/flow-software-architecture.png)
 
+Apache APISIX 是一个动态、实时、高性能的云原生 API 网关。它构建于 NGINX + ngx_lua 的技术基础之上，充分利用了 LuaJIT 所提供的强大性能。
```
[the remark this answers is not in the document]
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md:28]: Nice

[the remark this answers is not in the document]
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

File: docs/zh/latest/architecture-design/apisix.md (hunk 7 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

File: docs/zh/latest/architecture-design/apisix.md (hunk 8 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

File: docs/zh/latest/architecture-design/apisix.md (hunk 9 above)
Comment (bzp2010) [reply docs/zh/latest/architecture-design/apisix.md]: updated

**Review by SylviaBABY — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**